src/scripts/ui/MarketBuildingBody.tsx

```diff
@@ -16,6 +17,40 @@ import { BuildingStorageComponent } from "./BuildingStorageComponent";
 import { BuildingUpgradeComponent } from "./BuildingUpgradeComponent";
 import { BuildingWorkerComponent } from "./BuildingWorkerComponent";
 import { FormatNumber } from "./HelperComponents";
+import type { ResourceDefinitions } from "../../../shared/definitions/ResourceDefinitions";
+
+interface IMarketFiltersProps {
+   resourceFilterParams: {resourceFilter: string, setResourceFilter: React.Dispatch<React.SetStateAction<string>>},
```

Comment:
`React.Dispatch<React.SetStateAction<string>>` this type signature does not look right. It should be simplified.

Reply:
Simplified the types.